Tribler/Core/Modules/restapi/multichain_endpoint.py

```diff
@@ -294,26 +289,63 @@ def render_GET(self, request):
             return MultichainNetworkEndpoint.return_error(request, message="focus_node parameter empty")
 
         focus_node = request.args["focus_node"][0]
-        if isinstance(focus_node, basestring):
-            if request.args["focus_node"][0] == "self":
-                if mc_community.persistence.dummy_setup:
-                    focus_node = "00"
-                else:
-                    focus_node = hexlify(mc_community.my_member.public_key)
-            else:
-                focus_node = request.args["focus_node"][0]
-        else:
+        if not isinstance(focus_node, basestring):
             return MultichainNetworkEndpoint.return_error(request, message="focus_node was not a string")
 
-        neighbor_level = 1
-        # Note that isdigit() checks if all chars are numbers, hence negative numbers are not possible to be set
-        if "neighbor_level" in request.args and len(request.args["neighbor_level"]) > 0 and \
-                request.args["neighbor_level"][0].isdigit():
-            neighbor_level = int(request.args["neighbor_level"][0])
+        focus_node = self.get_focus_node(focus_node)
 
-        nodes, edges = mc_community.get_graph(unhexlify(focus_node), neighbor_level)
+        neighbor_level = self.get_neighbor_level(request.args)
+
+        nodes, edges = self.get_multi_chain_community().get_graph(unhexlify(focus_node), neighbor_level)
 
         return json.dumps({"focus_node": focus_node,
                            "neighbor_level": neighbor_level,
                            "nodes": nodes,
                            "edges": edges})
+
+    def use_dummy_data(self, dataset):
+        """
+        Set a dummy database for the rest of this run.
+
+        Note that once a dummy database is set, this dummy database is used until the software is restarted.
+        :param dataset: the dummy dataset type
+        :return:
+        """
+        if isinstance(dataset, basestring):
+            if dataset == "static" or dataset == "random":
+                mc_community = self.get_multi_chain_community()
+                if dataset == "static":
+                    mc_community.persistence.use_dummy_data(use_random=False)
+                elif dataset == "random":
+                    mc_community.persistence.use_dummy_data(use_random=True)
+
+    def get_focus_node(self, focus_node):
+        """
+        Set the focus node.
+
+        :return: the focus node value
+        """
+        if focus_node == "self":
+            mc_community = self.get_multi_chain_community()
+            if mc_community.persistence.dummy_setup:
+                focus_node = "00"
+            else:
+                focus_node = hexlify(mc_community.my_member.public_key)
+        return focus_node
+
+    @staticmethod
+    def get_neighbor_level(arguments):
```

Comment:
Why is this a static method and why do you give it the full list of arguments where the other methods only get the specific argument? (get_focus_node gets request.args["focus_node"], this one only gets request.args)